tests/client/client_test.go

```diff
@@ -188,6 +190,125 @@ func (s *clientTestSuite) TestLeaderTransfer(c *C) {
 	wg.Wait()
 }
 
+func (s *clientTestSuite) TestTSOAllocatorLeader(c *C) {
```

Comment:
I think we can use failpoint to inject error for the certain dc to verify the error handling.

Reply:
I think maybe we can refine the unit tests later by a new pr after merging this one.